components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/bean/SSOAgentConfig.java

```diff
@@ -192,8 +216,89 @@ public KeyStore getKeyStore() throws org.wso2.carbon.identity.sso.agent.exceptio
     public void setKeyStore(KeyStore keyStore) {
         this.keyStore = keyStore;
     }
+
     public void initConfig(Properties properties) throws SSOAgentException {
 
+        String decodedPassword;
+        boolean isReadpassword = false;
+        char[] password = null;
+
+        // Get copy of properties for looping in order to avoid ConcurrentModificationException.
+        Properties copyOfProperties = new Properties();
+        copyOfProperties.putAll(properties);
+
+        // Path of the password file.
+        String filePath = System.getProperty("catalina.home") + SSOAgentConstants.SSOAgentConfig.PASSWORD_FILEPATH;
+
+        // Looping through properties to check the encrypted property value by checking the prefix Enc:.
+        for (Map.Entry<Object, Object> entry : copyOfProperties.entrySet()) {
+            if (String.valueOf(entry.getValue()).startsWith("Enc:")) {
+                if (!System.getProperty(ARGUMENT).contains("password")) {
+
+                    // Check whether the password has been already read.
+                    if (!isReadpassword) {
+                        Path path = Paths.get(filePath);
+                        try (BufferedReader reader = Files.newBufferedReader(path, Charset.forName("UTF-8"))) {
+                            StringBuilder currentLine = new StringBuilder();
+
+                            // Read the password from the password file.
+                            currentLine.append(reader.readLine());
+                            password = new char[currentLine.length()];
+                            if (currentLine.length() > 0) {
+                                currentLine.getChars(0, currentLine.length(), password, 0);
+                            }
+                            isReadpassword = true;
+                            if (Files.deleteIfExists(path)) {
+                                LOGGER.info("Successfully deleted the file " + path);
```

Comment:
If we say `"Deleted the temporary password file at " + path`, that'll be more meaningful.